File: Plugins/Aspose_OCR_Java_for_Ruby/lib/asposeocrjava/OCR/detectreading.rb
module Asposeocrjava
  module DetectReading
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Initialize an instance of OcrEngine
        ocr_engine = Rjb::import('com.aspose.ocr.OcrEngine').new

        # Set the Image property by loading the image from file path location
        ocr_engine.setImage(Rjb::import('com.aspose.ocr.ImageStream').fromFile(data_dir + 'ocr.png'))

        # Set the DetectReadingOrder to true
        ocr_engine.getConfig().setDetectReadingOrder(true)

        # Process the image
        if ocr_engine.process()
            # Display the result
            puts "Text: " + ocr_engine.getText().to_string
        end
    end
  end
end

File: Plugins/Aspose_OCR_Java_for_PHP/src/aspose/ocr/WorkingWithOCR/DetectReading.php
<?php
namespace Aspose\OCR\WorkingWithOCR;

use com\aspose\ocr\OcrEngine as OcrEngine;
use com\aspose\ocr\ImageStream as ImageStream;

class DetectReading{
    public static function run($dataDir=null){

        # Initialize an instance of OcrEngine
        $ocr_engine = new OcrEngine();

        # Set the Image property by loading the image from file path location
        $imageStream=new ImageStream();
        $ocr_engine->setImage($imageStream->fromFile($dataDir . 'ocr.png'));

        # Set the DetectReadingOrder to true
        $ocr_engine->getConfig()->setDetectReadingOrder(true);

        # Process the image
        if ($ocr_engine->process()) {
            # Display the result
            print "Text: " . (string)$ocr_engine->getText().PHP_EOL;
        }
    }
}
?>